### DISCUSSION TITLE
Add something helps test in prod environment

### DISCUSSION THREAD

**Discussion by lz1998** [tokio-rs/axum#712] [Ideas]
There are two environments in our company, `test` and `prod`.
- `test`: We can connect database from our own computer.
- `prod`: We can't connect database from our own computer.

I can write a unit test to test whether handler works in `test` environment, but it's not convenient to write a test in `prod` environment. Is it possible to provide some helpers to convert `Json<T>` to `reqwest::Request`? It can helps developers test more conveniently.

Handler:
```rust
async fn handle_something(Json(req): axum::Json<Req>) -> Json<Resp> {...}

```
Test:
```rust
#[cfg(test)]
mod tests{
    use super::*;

    #[tokio::test]
    async fn test_handle_somethind(){
        let req = Req::default();
         // TODO how to switch between "unit test" and "online test" conveniently?
        let resp = handle_something(req).await;
        println("{:?}" ,resp);
    }
}
```

**Comment by davidpdrsn** [comment]
Are you trying to write a test that connects to your production server? I really would not recommend doing that since you want your tests to be deterministic.

If you really still want yo you'll have to construct requests manually with reqwest. I don't think adding code to axum to make that easier is necessary since axum is a server library and not a client.
